host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -146,21 +200,50 @@ defmodule HostCore.Providers.ProviderModule do
 
   @impl true
   def handle_call(:halt_cleanup, _from, state) do
-    # Elixir cleans up ports, but it doesn't always clean up the OS process it created
-    # for that port. TODO - find a clean, reliable way of killing these processes.
-    if state.os_pid != nil do
-      System.cmd("kill", ["-9", "#{state.os_pid}"])
-    end
+    Logger.debug("Provider termination requested manually")
+
+    Tracer.with_span "Terminate provider instance", kind: :server do
+      case HostCore.Nats.safe_req(
+             HostCore.Nats.rpc_connection(state.lattice_prefix),
+             "wasmbus.rpc.#{state.lattice_prefix}.#{state.public_key}.#{state.link_name}.shutdown",
+             Jason.encode!(%{host_id: state.host_id}),
```

Comment:
Remind me why we put the host ID on the body instead of the topic? Was it for backwards-compatibility reasons? Feels out of place

Reply:
backwards compatibility. yes, it feels out of place, but doing this allowed "old" hosts and providers to continue to work